distributed/comm/tcp.py

```diff
@@ -436,9 +436,17 @@ async def connect(self, address, deserialize=True, **connection_args):
         kwargs = self._get_connect_args(**connection_args)
 
         try:
-            stream = await self.client.connect(
-                ip, port, max_buffer_size=MAX_BUFFER_SIZE, **kwargs
-            )
+            # server_hostname option (for SNI) only works with tornado.iostream.IOStream
+            if "server_hostname" in kwargs:
+                plain_stream = await self.client.connect(
+                    ip, port, max_buffer_size=MAX_BUFFER_SIZE
+                )
+                stream = await plain_stream.start_tls(False, **kwargs)
```

Comment:
What happens if connect succeeds but start_tls fails (eg SSLCertVerificationError) does the plain_stream stay open indefinitely?

Reply:
Ah tornado handles this https://github.com/tornadoweb/tornado/blob/8ebbfea6af3f8fd829090025aecbe09f3fe65662/tornado/iostream.py#L1277